source/slang/slang-check-expr.cpp

```diff
@@ -4425,7 +4425,7 @@ namespace Slang
         expr->base = CheckProperType(expr->base);
         if (as<ErrorType>(expr->base.type))
             expr->type = expr->base.type;
-        auto ptrType = m_astBuilder->getPtrType(expr->base.type);
+        auto ptrType = m_astBuilder->getPtrType(expr->base.type, AddressSpace::UserPointer);
```

Comment:
Why do we use "UserPointer" here?

Reply:
A user defined pointer type in user code, such as `T*`, should be represented as a user-defined pointer in the IR to differentiate themselves from compiler-internal pointer types used for variables (`IROp_Var` etc.).

This will allow us to easily identify these user defined pointers and translate them to pointers in PhysicalStorageBufferClass address space when we get to spirv legalization.